products/views.py

```diff
@@ -1,3 +1,61 @@
-from django.shortcuts import render
+from django.db.models import Q, Count
+from products.models import Type, Product, Tag, TagProduct
+
+from django.http import JsonResponse
+from django.views import View
+
+
+class ProductCategoryView(View):
+    def get(self, request, category_id):
+        try:
+            types = Type.objects.filter(category=category_id)
+
+            result = [{
+                'name'         : type.name,
+                'thumbnail_url':type.thumbnail_image_url
+            } for type in types]
+
+            return JsonResponse({'message': result}, status=200)
+
+        except KeyError:
+            return JsonResponse({'message': 'invalid key'}, status=400)
```

Comment:
API를 살펴보니 `TypeView`가 더 어울리는 이름인 것 같은데 한번만 다시 확인해주세요!
```suggestion
class TypeView(View):
    def get(self, request, category_id):
        try:
            types = Type.objects.filter(category=category_id)

            result = [{
                'name'          : type.name,
                'thumbnail_url' : type.thumbnail_image_url
            } for type in types]

            return JsonResponse({'message': result}, status=200)

        except KeyError:
            return JsonResponse({'message': 'invalid key'}, status=400)
```

Reply:
TypeView로 네이밍 수정하도록 하겠습니다!